prow/repoowners/endpoint.go

```diff
@@ -0,0 +1,77 @@
+package repoowners
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+type OwnersServer struct {
+	OwnersClient *Client
+}
+
+type ownersRequest struct {
+	Org        string   `json:"org_name"`
+	Repo       string   `json:"repo_name"`
+	BaseCommit string   `json:"baseCommit"`
+	Paths      []string `json:"filepaths"`
+}
+
+type ownersResponse struct {
+	ConfigMap map[string]*Config `json:"owners,omitempty"`
+}
+
+func (s *OwnersServer) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if r.Header.Get("Content-Type") != "application/json" {
+		w.WriteHeader(http.StatusNotAcceptable)
+		return
+	}
+
+	rawBody, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	} else if len(rawBody) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	reqBody := &ownersRequest{}
+
+	if err = json.Unmarshal(rawBody, reqBody); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	ownerIfc, err := s.OwnersClient.LoadRepoOwners(reqBody.Org, reqBody.Repo, reqBody.BaseCommit)
+
+	if err != nil || ownerIfc == nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	requestedOwners := &ownersResponse{
+		ConfigMap: map[string]*Config{},
+	}
+
+	if len(reqBody.Paths) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+
+	for _, path := range reqBody.Paths {
+		requestedOwners.ConfigMap[path] = &Config{}
+		requestedOwners.ConfigMap[path].Approvers = ownerIfc.Approvers(path).List()
+		requestedOwners.ConfigMap[path].RequiredReviewers = ownerIfc.RequiredReviewers(path).List()
+		requestedOwners.ConfigMap[path].Reviewers = ownerIfc.Reviewers(path).List()
+		requestedOwners.ConfigMap[path].Labels = ownerIfc.FindLabelsForFile(path).List()
+
+	}
+	if rawResultBody, err := json.Marshal(requestedOwners.ConfigMap); err == nil {
```

Comment:
This provokes the `err` variable to shadowed. You can predeclare and `rawResultBody` with `var ...` and use a simple assignment `=` instead of a declare-assignment `:=`.